docs/tutorials/local-llm-setup.md

```diff
@@ -89,6 +89,41 @@ as the `chat_model` param in the `OpenAIGPTConfig` as in the previous section.
 When a script supports it, you can also pass in the model name via
 `-m ollama/dolphin-mixtral-gguf`
 
+## Setup llama.cpp with a GGUF model from HuggingFace
+
+See `llama.cpp`'s [GitHub page](https://github.com/ggerganov/llama.cpp/tree/master) for build and installation instructions.
+
+After installation, begin as above with downloading a GGUF model from HuggingFace; for example, the quantized `Qwen2.5-Coder-7B` from [here](https://huggingface.co/Qwen/Qwen2.5-Coder-7B-Instruct-GGUF); specifically, [this file](https://huggingface.co/Qwen/Qwen2.5-Coder-7B-Instruct-GGUF/blob/main/qwen2.5-coder-7b-instruct-q2_k.gguf).
+
+Now, the server can be started with `llama-server -m qwen2.5-coder-7b-instruct-q2_k.gguf`.
+
+In addition, your `llama.cpp` may be built with support for simplified management of HuggingFace models (specifically, `libcurl` support is required); in this case, `llama.cpp` will download HuggingFace models to a cache directory, and the server may be run with:
+```bash
+llama-server \
+      --hf-repo Qwen/Qwen2.5-Coder-7B-Instruct-GGUF \
+      --hf-file qwen2.5-coder-7b-instruct-q2_k.gguf
+```
+
+To use the model with Langroid, specify `llamacpp/localhost:{port}` as the `chat_model`; the default port is 8080.
+
+## Setup vLLM with a model from HuggingFace
```

Comment:
Any restrictions on where we can use vllm, e.g. linux only, and/or is this only relevant when using a GPU?

Reply:
Right now it's linux only. I don't think VLLM has any support for accelerators on M series chips; there's a PR for ARM support right now but that would still be CPU only. Currently, CPU only mode seems to require x86. 

With the new PR it may run on macOS, but llama.cpp would likely still be preferred in that case (as it supports the accelerators, etc.).

Reply:
Another note is that installation with backends other than NVIDIA GPUs requires rebuilding from source.